components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Publisher/src/Runtime/PublishServicesConfig.cs

```diff
@@ -70,8 +70,16 @@ public class PublishServicesConfig : ConfigBase, IPublishServicesConfig {
         public int DefaultMaxEgressMessageQueue {
             get {
                 var queueSize = GetIntOrDefault(kDefaultMaxEgressMessageQueue,
-                    () => GetIntOrDefault(PcsVariable.PCS_MAX_EGRESS_MESSAGE_QUEUE,
-                    () => _DefaultMaxEgressMessageQueue));
+                    () => GetIntOrDefault(PcsVariable.PCS_DEFAULT_PUBLISH_MAX_EGRESS_MESSAGE_QUEUE,
+                    () => -1));
+
+                if (queueSize == -1) {
+                    // Fallback to deprecated option,
+                    // use PCS_DEFAULT_PUBLISH_MAX_EGRESS_MESSAGE_QUEUE instead.
+                    queueSize = GetIntOrDefault(kDefaultMaxEgressMessageQueue,
+                        () => GetIntOrDefault(PcsVariable.PCS_DEFAULT_PUBLISH_MAX_OUTGRESS_MESSAGES,
```

Comment:
Would you mind also writing a test that validates that it is applied ? Something similar to `PublishServicesConfigTest2Async()` in `components\opc-ua\src\Microsoft.Azure.IIoT.OpcUa.Publisher\tests\Clients\PublisherJobClientTests.cs`.

Reply:
I cannot test it like PublishServicesConfigTest2Async because the deprecated keys are mapped to the new ones before they reach the "configuration" object.

Reply:
Yes, you are correct about the mapping. What I mean is that you can test that when `PCS_DEFAULT_PUBLISH_MAX_OUTGRESS_MESSAGES` is set to some value `x` that value is actually applied. For example, you can check that `Confi.DefaultMaxEgressMessageQueue` has that value `x` or that job definition has that value `x` when job is created. If you want to, you can mark this conversation as resolved as well, since we know that general mechanism works.

Reply:
What I meant was that I cannot test the params maxoutgressmessages/maxoutgressmessages in the same fashion, but I can test the PCS variables. Added two tests for that.